### PR TITLE
#112 Adding feature to insert document when overwrite and putIfAbsent…

### PR DESCRIPTION
**Pull request by vipulkashyap111**
… true

### CODE REVIEW

**Comment by vipulkashyap111** [comment]
@jasonpet @rabbah This is for #112. Added putIfAbsent flag which if set to true along with overwrite=true will do an insertion operation if the document does not already exists(404)

**Comment by rabbah** [comment]
Thanks for submitting the PR. Do you have an Apache ICLA on file? https://github.com/apache/incubator-openwhisk-package-cloudant/blob/master/CONTRIBUTING.md

**Comment by rabbah** [comment]
It would be nice to add a unit test similar to the one here https://github.com/apache/incubator-openwhisk-package-cloudant/blob/master/tests/src/test/scala/system/packages/CloudantDatabaseActionsTests.scala#L290.

**Comment by jasonpet** [comment]
I can add a test and update the documentation.  I will open a separate PR

**Comment by vipulkashyap111** [comment]
Added a new variable to provide backward compatibility. I'll update the code as per @csantanapr review comments, do @jasonpet and @rabbah agree?

**Comment by rabbah** [comment]
@csantanapr I generally agree the intended behavior is to just update. The concern was about breaking existing code. 

There are create and update actions which do the more conservative behavior. So we can treat this as a bug, remove the parameter and just do the update on 404. I'm OK with that too.

**Comment by jasonpet** [comment]
I guess only a test is needed now since we are not adding a new parameter.

**Comment by vipulkashyap111** [comment]
Removed putIfAbsent flag, document is inserted if overwrite=true and it does not already exists in the database

File: actions/database-actions/write-document.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ function main(message) {
 /**
  * If id defined and overwrite is true, checks if doc exists to retrieve version
  * before insert. Else inserts.
+ * If
```
Comment (rabbah) [inline actions/database-actions/write-document.js]: dangling comment.

File: actions/database-actions/write-document.js [hunk 2]
```diff
@@ -62,8 +63,13 @@ function insertOrUpdate(cloudantDb, overwrite, doc) {
                                reject(err);
                             });
                     } else {
-                        console.error('error', error);
-                        reject(error);
+                        if(error.statusCode === 404) {
+                            // If document not found, insert it
+                            return insert(cloudantDb, doc);
```
Comment (jasonpet) [inline actions/database-actions/write-document.js]: the promise that contains the cloudant "get" is never resolved for this path

Comment (rabbah) [reply actions/database-actions/write-document.js]: good catch, looks like insert creates it own promise.

Comment (vipulkashyap111) [reply actions/database-actions/write-document.js]: let me update that

Comment (vipulkashyap111) [reply actions/database-actions/write-document.js]: updated

**Comment by rabbah** [comment]
@csantanapr changes updated, can you update your review?

**Pull request merged**